analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -195,7 +195,8 @@ private ProgramState ProcessInvocation(SymbolicContext context, IInvocationOpera
                 }
             }
             return ProcessAddMethod(context.State, invocation.TargetMethod, instance)
-                ?? ProcessRemoveMethod(context.State, invocation.TargetMethod, instance);
```

Comment:
Nitpick:
Maybe you can retrieve once the `invocation.TargetMethod.Name` and pass it to all the calls as an argument.